core/rawdb/database.go

```diff
@@ -229,11 +222,11 @@ func NewDatabaseWithFreezer(db ethdb.KeyValueStore, freezer string, namespace st
 		offset = ReadOffSetOfCurrentAncientFreezer(db)
 	}
 
-	frdb.offset = offset
-
-	// Some blocks in ancientDB may have already been frozen and been pruned, so adding the offset to
-	// reprensent the absolute number of blocks already frozen.
-	frdb.frozen += offset
+	// Create the idle freezer instance
+	frdb, err := newFreezer(freezer, namespace, readonly, offset, freezerTableSize, FreezerNoSnappy)
+	if err != nil {
+		return nil, err
```

Comment:
better add a error log to print error info

Reply:
There is a print error log in the upper layer, and there will be duplicates when adding a log here